packages/patternfly-3/react-console/src/VncConsole/VncActions.js

```diff
@@ -1,30 +1,60 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import { MenuItem, Button, helpers } from 'patternfly-react';
 
 const { Dropdown } = Button;
 const { noop } = helpers;
 
-const VncActions = ({ textSendShortcut, textCtrlAltDel, onCtrlAltDel }) => (
-  <Dropdown bsStyle="default" title={textSendShortcut} id="console-send-shortcut" onClick={noop}>
-    <MenuItem eventKey="1" onClick={onCtrlAltDel}>
-      {textCtrlAltDel}
-    </MenuItem>
-  </Dropdown>
-);
+const VncActions = ({
+  textSendShortcut,
+  textCtrlAltDel,
+  textDisconnect,
+  onCtrlAltDel,
+  onDisconnect,
+  portalToolbarTo
+}) => {
+  const toolbar = (
+    <div>
+      <Dropdown bsStyle="default" title={textSendShortcut} id="console-send-shortcut" onClick={noop}>
+        <MenuItem eventKey="1" onClick={onCtrlAltDel}>
+          {textCtrlAltDel}
+        </MenuItem>
+      </Dropdown>
+      <Button bsStyle="default" onClick={onDisconnect}>
+        {textDisconnect}
+      </Button>
+    </div>
+  );
+
+  if (!portalToolbarTo) {
+    return toolbar;
+  }
+  return (
+    document.getElementById(portalToolbarTo) && ReactDOM.createPortal(toolbar, document.getElementById(portalToolbarTo))
```

Comment:
I don't think we should perform such manipulations. I believe change in the component composition should be preferred over this approach.

Reply:
Yes, but in that case appear problem with customization. For example, in VM Portal we need only one toolbar not two and we can't `Send Key` dropdown move to our toolbar, because console is fully under control of `VncConsole` component.

And I think this is most suitable solution.